solr/core/src/java/org/apache/solr/handler/admin/api/RestoreCollectionAPI.java

```diff
@@ -144,10 +144,14 @@ public SubResponseAccumulatingJerseyResponse restoreCollection(
       throw remoteResponse.getException();
     }
 
+    if (requestBody.async != null) {
+      response.requestId = requestBody.async;
+      return response;
```

Comment:
I am curious to know if there is any reason to return here and skip the 2 lines below?